### PR TITLE
sent and received all devices location through wamp protocols

### PR DESCRIPTION
**Pull request by shakeebkhan66**

### CODE REVIEW

File: lib/model/latlong_model.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import "dart:convert";
```
Comment (muzzammilshahid) [inline lib/model/latlong_model.dart:1]: We are not using this file?

File: lib/res/wamp_util.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import "package:location_tracker/res/constants.dart";
```
Comment (muzzammilshahid) [inline lib/res/wamp_util.dart:1]: This file is not needed. Now we have static data, and this can be simplified.

File: lib/res/constants.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import "package:flutter/material.dart";
+import "package:flutter_spinkit/flutter_spinkit.dart";
+
+/// Serializers ///
+const jsonSerializer = "JSON";
+const cborSerializer = "CBOR";
+const msgPackSerializer = "MsgPack";
```
Comment (muzzammilshahid) [inline lib/res/constants.dart:7]:
```suggestion
/// Serializers ///
```

**Review by muzzammilshahid — CHANGES_REQUESTED** [review]

File: lib/controllers/live_location_cont.dart [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  RxMap<String, Polyline> polylines = <String, Polyline>{}.obs;
+  GoogleMapController? mapController;
+  StreamSubscription<Position>? _positionStreamSubscription;
+  Rx<DateTime> lastPublishTime = DateTime.now().obs;
+  String deviceName = "";
+  Session? session;
+
+  Future<void> mobileModelName() async {
+    final plugin = DeviceName();
+    deviceName = (await plugin.getName())!;
```
Comment (asimfarooq5) [inline lib/controllers/live_location_cont.dart:27]: Add null check for safety: 
deviceName = await plugin.getName() ?? "Unknown Device";

File: lib/controllers/live_location_cont.dart [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    await _publish(30.19182, 71.44313, deviceName);
```
Comment (asimfarooq5) [inline lib/controllers/live_location_cont.dart:28]: Avoid hardcoding coordinates.

**Review by asimfarooq5 — CHANGES_REQUESTED** [review]

File: lib/controllers/live_location_cont.dart [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+  Future<void> mobileModelName() async {
+    final plugin = DeviceName();
+    deviceName = (await plugin.getName())!;
+    print("deviceName $deviceName");
+    await _publish(30.19182, 71.44313, deviceName);
+  }
+
+  @override
+  Future<void> onInit() async {
+    print("hello");
```
Comment (asimfarooq5) [inline lib/controllers/live_location_cont.dart]: Remove debug code.

**Review by asimfarooq5 — CHANGES_REQUESTED** [review]

**Review by asimfarooq5 — APPROVED** [review]

**Review by muzzammilshahid — CHANGES_REQUESTED** [review]

**Review by muzzammilshahid — APPROVED** [review]

**Pull request merged**